api/v1alpha1/provisioning_types.go

```diff
@@ -38,6 +38,16 @@ const (
 	ProvisioningSingletonName = "provisioning-configuration"
 )
 
+// BootIsoSource is the origin of the boot iso image
+// +kubebuilder:validation:Enum=local;http
+type BootIsoSource string
+
+// BootIsoSource values
+const (
+	BootIsoSourceLocal BootIsoSource = "local"
+	BootIsoSourceHttp  BootIsoSource = "http"
```

Comment:
Did we bike shed on the names we want to use?

`cache` and `direct` have more obvious meanings to me than `local` and `http`.

Reply:
this makes sense, although the values of "local" and "http" can't be changed as they're interpreted by Ironic
or maybe I misunderstood and you meant to change BootIsoSourceLocal and BootIsoSourceHttp ?

Reply:
I think it makes sense to name them after the ironic values